test/gtest/graphapi_conv_bias_res_add_activ_fwd.cpp

```
@@ -45,13 +46,7 @@
 namespace gr = miopen::graphapi;
 namespace conv_graph_api_test {
 
-bool IsTestSupportedForDevice()
-{
-    using e_mask = enabled<Gpu::gfx94X, Gpu::gfx103X, Gpu::gfx110X>;
-    // gfx120X is not enabled due to WORKAROUND_SWDEV_479810
-    using d_mask = disabled<Gpu::None>;
-    return ::IsTestSupportedForDevMask<d_mask, e_mask>();
-}
+bool IsTestSupportedForDevice() { return ::IsDeviceSupportedForCK(); }
```

How do we know that `graphapi` use only `CK`? There should be no dependency here.

---

The graphAPI implementation in this case is using the fused CK solver.
I agree that it's not really clear, but not really sure how else we can avoid this issue.

---

I think we still need this check either way for the tests, but maybe it's better to have it fail the matches check for the pattern as well on unsupported devices?

```cpp
bool matches(const OpGraph* graph_ptr) const final
    {
        assert(graph_ptr);

        if(!isIsomorphic(*graph_ptr, getPatternGraph()))
        {
            return false;
        }

        return hasBiasNode(*graph_ptr);
    }
```

At least then it'll fail earlier, and in the future if we add additional patterns that match (like a generic one), then it will only match a valid one.

If we go that route, then I guess we would want to rename the executor to indicate it's CK specific, and then in the matches we can check if CK is supported since we only have a CK executor available. Later on if we add additional executors for this pattern, then we can remove these checks.

---

My point is that there should be a check for devices that work with this test at the moment. I am not aware of current graphapi implementation, but this is a general test, it should not have any backend bindings.

---

Oh you mean you would prefer to have just the checks inside the test, and not have the shared file with fused?

I combined it since they both have the same requirements, but could split it back out if you think that's better.

---

Yes, inside _this_ test. Also let's keep that file for CK.

---

Do these changes match what you were expecting?